src/test/java/org/broadinstitute/barclay/argparser/TaggedArgumentTest.java

```diff
@@ -0,0 +1,301 @@
+
+    @SuppressWarnings("unchecked")
+    @DataProvider(name = "GoodTaggedArguments")
+    public Object[][] goodTaggedArguments() {
+        return new Object[][][]{
+                //TODO: file args, tagged scalar/enum (?), intermix tagged with positional args
+                {
+                        // single value in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "tumor.bam")},
+                        new KeyValuePair[][]{{}}
+                },
+                {
+                        // single value in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "/tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "/tumor.bam")},
+                        new KeyValuePair[][]{{}}
+                },
+                {
+                        // single tagged value in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "gcs://my/tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "gcs://my/tumor.bam")},
+                        new KeyValuePair[][]{{}}
+                },
+                {
+                        // single value in a collection, no attributes, full name
+                        new String[]{"--tFullName:tumor", "tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "tumor.bam")},
+                        new KeyValuePair[][]{{}}
+                },
+                {
+                        // single tagged value in a collection, two attributes, short name
+                        new String[]{"--t:tumor,truth=true,training=false", "tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "tumor.bam")},
+                        new KeyValuePair[][]{
+                                {
+                                        KeyValuePair.of("truth", "true"),
+                                        KeyValuePair.of("training", "false")
+                                }
+                        }
+                },
+                {
+                        // single tagged value in a collection, two attributes, short name
+                        new String[] {"--t:tumor,truth=false,training=true", "tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "tumor.bam")},
+                        new KeyValuePair[][]{
+                                {
+                                        KeyValuePair.of("truth", "false"),
+                                        KeyValuePair.of("training", "true")
+                                }
+                        }
+                },
+                {
+                        // two tagged values in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "tumor.bam", "--t:normal", "normal.bam"},
```

Comment:
Add test case: two args, one tagged, one untagged, and assert that the untagged one does not get tagged